### PR TITLE
Referrer Policy: inheritance with javascript: URLs and document.write()

### PR DESCRIPTION
**Pull request by annevk**
This cleans up some existing tests and adds new tests with results I expect from reading the specification, but do not see confirmed in implementations.

### CODE REVIEW

File: referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    expected: self.origin + "/"
+  },
+  {
+    srcDocPolicy: `<meta name="referrer" content="no-referrer">`,
+    expected: undefined
+  }
+].forEach(({ srcDocPolicy, expected }) => {
+  promise_test(t => {
+    return new Promise(resolve => {
+      window.addEventListener("message", t.step_func(msg => {
```
Comment (mikewest) [inline referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html]: I don't think we need `step_func` in a `promise_test`, do we?

**Review by mikewest — APPROVED** [review]
This look right to me. I wonder if they pass in browsers? Let's find out. :)

One style nit, otherwise LGTM.

File: referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html (hunk 1 above)
Comment (annevk) [reply referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html]: This is what https://web-platform-tests.org/writing-tests/testharness-api.html#promise-tests suggests, but if I remove the wrappers I end up with a harness failure.

**Comment by annevk** [comment]
Both Chrome and Firefox have (different) failures in iframe-inheritance-javascript.html and iframe-inheritance-javascript-child.html. Firefox also fails iframe-inheritance-document-write.html.

The test is assuming that the policy of the document the URL executes in is the one supposed to be reused (i.e., the target not the source).

File: referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html (hunk 1 above)
Comment (annevk) [reply referrer-policy/generic/inheritance/iframe-inheritance-javascript-child.html]: That was a mistake in the documentation, https://github.com/web-platform-tests/wpt/pull/21269 fixed it.

**Comment by annevk** [comment]
Filed https://bugs.chromium.org/p/chromium/issues/detail?id=1043895 and https://bugzilla.mozilla.org/show_bug.cgi?id=1610450.

**Pull request merged**